packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/BlocksToolbar.tsx

```diff
@@ -496,23 +516,132 @@ const LinkButton = ({ disabled }: { disabled: boolean }) => {
     insertLink(editor, { url: '' });
   };
 
+  const label = {
+    id: 'components.Blocks.link',
+    defaultMessage: 'Link',
+  } as MessageDescriptor;
+
+  if (location === 'menu') {
+    return (
+      <Menu.Item onClick={addLink} disabled={isLinkDisabled()}>
+        {formatMessage(label)}
+      </Menu.Item>
+    );
+  }
+
   return (
     <ToolbarButton
       icon={Link}
       name="link"
-      label={{
-        id: 'components.Blocks.link',
-        defaultMessage: 'Link',
-      }}
+      label={label}
       isActive={isLinkActive()}
       handleClick={addLink}
       disabled={isLinkDisabled()}
     />
   );
 };
 
+interface ObservedToolbarItemProps {
+  index: number;
+  lastVisibleIndex: number;
+  setLastVisibleIndex: React.Dispatch<React.SetStateAction<number>>;
+  rootRef: React.RefObject<HTMLElement>;
+  children: React.ReactNode;
+}
+
+const ObservedToolbarItem = ({
```

Comment:
Would this be something we can abstract to share with the markdown editor as well?